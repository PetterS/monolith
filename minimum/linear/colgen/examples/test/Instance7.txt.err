Number of days: 28
-- Shift E, 480 min.
-- Shift D, 480 min.
-- Shift L, 480 min.
Number of shifts: 3
-- Shift 0. OK after = [0, 1, 2] forbidden_after = []
-- Shift 1. OK after = [1, 2] forbidden_after = [0]
-- Shift 2. OK after = [2] forbidden_after = [0, 1]
-- Staff  A: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 0].
-- Staff  B: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 0].
-- Staff  C: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 9].
-- Staff  D: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 0, 0].
-- Staff  E: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 9].
-- Staff  F: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 0, 9].
-- Staff  G: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 0, 0].
-- Staff  H: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 9].
-- Staff  I: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 0].
-- Staff  J: Consequtive limit [2, 5]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 9].
-- Staff  K: Consequtive limit [2, 6]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 9].
-- Staff  L: Consequtive limit [2, 6]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 9].
-- Staff  M: Consequtive limit [2, 6]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 9].
-- Staff  N: Consequtive limit [2, 6]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 9].
-- Staff  O: Consequtive limit [2, 6]. Time limit [7560, 8640]. Max weekends: 2. Max shifts: [28, 28, 0].
-- Staff  P: Consequtive limit [1, 5]. Time limit [3240, 4320]. Max weekends: 3. Max shifts: [0, 28, 4].
-- Staff  Q: Consequtive limit [1, 5]. Time limit [3240, 4320]. Max weekends: 3. Max shifts: [28, 28, 0].
-- Staff  R: Consequtive limit [1, 5]. Time limit [3240, 4320]. Max weekends: 3. Max shifts: [28, 28, 4].
-- Staff  S: Consequtive limit [1, 5]. Time limit [3240, 4320]. Max weekends: 3. Max shifts: [28, 28, 4].
-- Staff  T: Consequtive limit [1, 5]. Time limit [3240, 4320]. Max weekends: 3. Max shifts: [28, 28, 0].
Number of staff: 20
Read 104 shift on requests.
Read 64 shift off requests.
Read 84 cover constraints.
Dividing all minutes by 480
Setting up colgen problem...[  OK  ]
Creating initial rosters...[  OK  ]
-- Objective for initial columns: 8597
  Iter |       Objective      |    Problem   |    Pool   | Fixed  |                  Time                  | Frac. |
       |    frac.     rounded |   size  diff | size gen. |        |    fix       gen     solve   tot. cum. |       |
      1       8597       8597        0  188     188    0      0        n/a       n/a     n/a     n/a          0%
      2     5053.9       5331       45   45     233   45      0        n/a       n/a     n/a     n/a          1%
      3    4224.61       4627       79   34     267   34      0        n/a       n/a     n/a     n/a          8%
      4    3185.18       4408      110   31     300   33      0        n/a       n/a     n/a     n/a          9%
      5    2685.89       5023      151   41     339   39      0        n/a       n/a     n/a     n/a          8%
      6    2354.37       3793      184   33     373   34      0        n/a       n/a     n/a     n/a          2%
      7    2018.33       4825      215   31     418   45      0        n/a       n/a     n/a     n/a          0%
      8    1873.28       2984      226   11     450   32      0        n/a       n/a     n/a     n/a          0%
      9    1712.22       3607      251   25     483   33      0        n/a       n/a     n/a     n/a          7%
     10    1565.75       4415      253    2     526   43      0        n/a       n/a     n/a     n/a          9%
     11    1473.48       3620      256    3     556   30      0        n/a       n/a     n/a     n/a          2%
     12    1351.63       3202      244  -12     584   28      0        n/a       n/a     n/a     n/a          2%
     13    1312.91       3604      255   11     615   31      0        n/a       n/a     n/a     n/a          1%
     14    1267.75       3810      257    2     650   35      0        n/a       n/a     n/a     n/a          2%
     15    1219.68       3517      251   -6     683   33      0        n/a       n/a     n/a     n/a          3%
     16       1197       3513      254    3     719   36      0        n/a       n/a     n/a     n/a          4%
     17    1188.96       3410      259    5     748   29      0        n/a       n/a     n/a     n/a          3%
     18     1179.6       3108      269   10     788   40      0        n/a       n/a     n/a     n/a          1%
     19    1166.85       3592      275    6     824   36      0        n/a       n/a     n/a     n/a          0%
     20    1156.62       4190      285   10     856   32      0        n/a       n/a     n/a     n/a          8%
     21    1148.52       3475      277   -8     891   35      0        n/a       n/a     n/a     n/a          0%
     22    1143.45       3682      280    3     935   44      0        n/a       n/a     n/a     n/a          1%
     23    1133.48       4171      270  -10     976   41      0        n/a       n/a     n/a     n/a          1%
     24    1125.17       3341      261   -9    1014   38      0        n/a       n/a     n/a     n/a          2%
     25    1120.03       3545      257   -4    1048   34      0        n/a       n/a     n/a     n/a          2%
     26    1116.15       2934      244  -13    1081   33      0        n/a       n/a     n/a     n/a          4%
     27    1110.94       3322      238   -6    1122   41      0        n/a       n/a     n/a     n/a          4%
     28     1107.3       2921      227  -11    1152   30      0        n/a       n/a     n/a     n/a          6%
     29    1103.43       3525      219   -8    1189   37      0        n/a       n/a     n/a     n/a          6%
     30    1100.37       2918      210   -9    1214   25      0        n/a       n/a     n/a     n/a          8%
     31    1095.32       2609      221   11    1259   45      0        n/a       n/a     n/a     n/a          7%
     32    1092.12       2812      223    2    1296   37      0        n/a       n/a     n/a     n/a          6%
     33    1091.22       3316      221   -2    1337   41      0        n/a       n/a     n/a     n/a          7%
-- Fixed for [A: 18, B: 7, C: 7, D: 15, E: 6, F: 18, G: 18, H: 11, I: 12, J: 10, K: 7, L: 17, M: 12, N: 12, O: 10, P: 4, Q: 3, R: 7, S: 5, T: 6] at level 0.75.
-- Members remaining: (B, 2) (C, 3) (D, 2) (E, 3) (H, 2) (I, 2) (K, 3) (M, 2) (N, 2) (O, 2) (P, 5) (Q, 5) (S, 5) (T, 2) and total is 14.
     34    1640.79       2504       71 -150     248   16   1105        n/a       n/a     n/a     n/a          1%
     35     1362.1       2513       85   14     264   16      0        n/a       n/a     n/a     n/a          2%
     36    1165.92       1804      101   16     280   16      0        n/a       n/a     n/a     n/a          2%
     37     1112.1       2211      113   12     294   14      0        n/a       n/a     n/a     n/a          5%
     38    1091.42       2106      130   17     309   15      0        n/a       n/a     n/a     n/a          9%
     39    1088.76       2001      134    4     324   15      0        n/a       n/a     n/a     n/a          6%
     40    1087.47       1797      144   10     338   14      0        n/a       n/a     n/a     n/a          6%
     41    1086.82       1693      153    9     353   15      0        n/a       n/a     n/a     n/a          5%
-- Fixed for [C: 11, D: 3, E: 6, H: 1, I: 2, J: 8, K: 4, M: 4, N: 4, O: 3, P: 2, Q: 2, R: 2, S: 3, T: 3] at level 0.75.
-- Members remaining: (B, 3) (E, 2) (H, 3) (I, 4) (K, 2) (M, 2) (O, 3) (P, 4) (Q, 4) and total is 9.
     42     1288.5       2202       63  -90     239    9    123        n/a       n/a     n/a     n/a          4%
     43       1122       1491       73   10     249   10      0        n/a       n/a     n/a     n/a          4%
     44     1108.8       1491       80    7     258    9      0        n/a       n/a     n/a     n/a          6%
     45       1088       1595       83    3     266    8      0        n/a       n/a     n/a     n/a          9%
     46     1086.5       1997       89    6     274    8      0        n/a       n/a     n/a     n/a          8%
     47       1086       1997       87   -2     284   10      0        n/a       n/a     n/a     n/a          9%
-- Fixed for [E: 1, H: 1, I: 2, O: 1] at level 0.75.
-- Members remaining: (B, 2) (E, 2) (H, 2) (I, 2) (K, 2) (M, 2) (O, 2) (P, 2) (Q, 2) (S, 2) and total is 10.
     48       1086       1997       83   -4     270    8     22        n/a       n/a     n/a     n/a          9%
-- Fixed to zero for [A: 32, B: 33, C: 20, D: 58, E: 15, F: 37, G: 58, H: 12, I: 38, J: 9, K: 19, L: 17, M: 16, N: 23, O: 33, P: 44, Q: 35, R: 15, S: 28, T: 38] at level 0.05.
-- Fixing column 813 to 1 for member Q, was 0.5
-- Members remaining: (B, 2) (E, 2) (H, 2) (I, 2) (K, 2) (M, 2) (O, 2) (P, 2) (Q, 2) (S, 2) and total is 10.
     49       1222       1290       74   -9     272    7      1        n/a       n/a     n/a     n/a          0%
     50    1188.67       1390       73   -1     278    6      0        n/a       n/a     n/a     n/a          2%
     51       1188       1289       67   -6     284    6      0        n/a       n/a     n/a     n/a          4%
-- Fixed for [E: 2, Q: 4] at level 0.75.
-- Fixed to zero for [A: 3, B: 4, C: 2, E: 15, F: 1, G: 1, H: 10, I: 5, J: 5, K: 12, L: 13, M: 21, N: 9, O: 3, P: 6, Q: 13, R: 21, S: 18, T: 8] at level 0.05.
-- Members remaining: (B, 2) (E, 3) (H, 2) (I, 2) (K, 2) (M, 3) (O, 2) (P, 3) (S, 2) and total is 9.
     52       1187       1187       64   -3     266    4     16        n/a       n/a     n/a     n/a          0%
-- Fixed for [B: 11, E: 1, H: 4, I: 2, K: 5, M: 2, O: 3, P: 3, S: 1] at level 0.75.
-- Fixed to zero for [B: 2, E: 3, F: 1, H: 2, I: 1, J: 2, K: 1, L: 2, M: 4, N: 2, O: 4, Q: 4, R: 3, S: 4, T: 1] at level 0.05.
-- Members remaining: and total is 0.
     53       1187       1187       23  -41     191    0     75        n/a       n/a     n/a     n/a          0%
-- Stopping. Objective change is 0 with an integer feasible solution.
Colgen done.
Elapsed time : n/a.
  Iter |       Objective      |    Problem   |    Pool   | Fixed  |                  Time                  | Frac. |
       |    frac.     rounded |   size  diff | size gen. |        |    fix       gen     solve   tot. cum. |       |
      1       1187       1187       23    0    1543    0      0        n/a       n/a     n/a     n/a          0%
      2       1187       1187      232  209    1586   43      0        n/a       n/a     n/a     n/a          0%
      3    1088.81       1390      648  416    1619   33      0        n/a       n/a     n/a     n/a          3%
      4    1080.08       1893      755  107    1662   43      0        n/a       n/a     n/a     n/a          0%
      5    1078.87       1996      783   28    1701   39      0        n/a       n/a     n/a     n/a          1%
      6    1076.87       2592      797   14    1732   31      0        n/a       n/a     n/a     n/a          0%
      7    1076.19       2586      631 -166    1762   30      0        n/a       n/a     n/a     n/a          3%
-- Fixed for [A: 13, B: 7, C: 15, D: 16, E: 8, F: 18, G: 16, H: 12, I: 14, J: 18, K: 6, L: 9, M: 18, N: 16, O: 12, P: 7, Q: 3, R: 6, S: 6, T: 9] at level 0.75.
-- Members remaining: (A, 2) (B, 3) (D, 2) (E, 3) (G, 2) (H, 3) (I, 3) (K, 5) (L, 2) (N, 2) (O, 2) (P, 3) (Q, 4) (S, 2) and total is 14.
      8     1263.8       1386      102 -529     331   13   1444        n/a       n/a     n/a     n/a          8%
      9    1232.17       1485      114   12     345   14      0        n/a       n/a     n/a     n/a          6%
     10    1190.38       1481      117    3     357   12      0        n/a       n/a     n/a     n/a          9%
     11       1153       1583      129   12     370   13      0        n/a       n/a     n/a     n/a          8%
     12    1137.35       1791      135    6     380   10      0        n/a       n/a     n/a     n/a          0%
     13    1132.48       1788      129   -6     391   11      0        n/a       n/a     n/a     n/a          0%
     14    1132.48       1788      139   10     401   10      0        n/a       n/a     n/a     n/a          8%
     15    1132.48       1788      134   -5     409    8      0        n/a       n/a     n/a     n/a          1%
-- Fixed for [A: 3, B: 3, C: 1, D: 1, E: 5, G: 2, H: 1, I: 1, K: 6, L: 8, N: 1, O: 2, Q: 2, R: 3, S: 3] at level 0.75.
-- Members remaining: (A, 3) (B, 2) (C, 2) (D, 2) (E, 3) (H, 4) (I, 3) (K, 2) (O, 3) (P, 5) (Q, 5) and total is 11.
     16       1183       1386       67  -67     276    6    139        n/a       n/a     n/a     n/a          3%
     17     1163.6       1285       77   10     284    8      0        n/a       n/a     n/a     n/a          0%
     18     1133.5       2093       77    0     291    7      0        n/a       n/a     n/a     n/a          4%
     19     1133.5       2093       85    8     296    5      0        n/a       n/a     n/a     n/a          2%
-- Fixed for [A: 1, C: 2, E: 1, H: 1, I: 1, Q: 2] at level 0.75.
-- Members remaining: (A, 2) (B, 2) (D, 2) (E, 2) (H, 2) (I, 2) (K, 2) (O, 2) (P, 2) (Q, 2) and total is 10.
     20    1150.33       1184       53  -32     254    4     46        n/a       n/a     n/a     n/a          7%
     21    1116.33       1183       63   10     259    5      0        n/a       n/a     n/a     n/a          5%
     22    1116.33       1184       69    6     264    5      0        n/a       n/a     n/a     n/a          4%
-- Fixed for [A: 1, B: 8, D: 1, E: 2, H: 3, I: 1, K: 4, O: 3, P: 2, Q: 2] at level 0.6075.
-- Members remaining: (I, 2) and total is 1.
     23       1183       1183       22  -47     191    1     74        n/a       n/a     n/a     n/a          0%
     24       1183       1183       23    1     191    0      0        n/a       n/a     n/a     n/a          0%
-- Stopping. Objective change is 0 with an integer feasible solution.
Colgen done.
Elapsed time : n/a.
Solution objectives: [1187, 1183].
