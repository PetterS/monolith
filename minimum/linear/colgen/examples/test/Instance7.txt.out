  A: D=D=D=D=D= . .D=D= . .D=D=D=D= . .E=D=D=D= . .D=D=D= . .
  B:  .E=E=E=D= . .D=D=D=D= . . .E=E=E= . .E=E=E=D=D= . .D=D=
  C:  . .E=E=L= . .D=D=L=L=L= . .D=D= . .E=E=E= . .D=D=D=L=L=
  D: E=E=E= . . . .E=E=E=E=E= . . . .E=E=E=E=E= . .E=E=E=E=E=
  E: D=D=D= . .E=D=L=L=L= . . . . .E=E=D=L= . . .L=L= . .D=D=
  F: L=L= . .E=E=E=E=E= . .E=E=E=L= . .L=L= . .E=E=E=E=E= . .
  G: E=E= . .E=E=E=E=E= . .E=E=E=E= . .E=E= . .E=E=E=E=E= . .
  H: E=E=E=E=E= . .D=D= . .D=L=L= . .D=D=D=D=L= . . .D=D= . .
  I: D=D=D=D=D= . . .E=D=D= . . .D=D= . .E=E=E=E=E= . .E=E=E=
  J:  . .D=D=D=D=D= . .E=D=D=L=L= . .D=D=D= . .D=L=L=L=L= . .
  K: E=D=D=D=D= . . . . . .E=E=E=E=D= . . .E=E=E=E=E=E= . . .
  L:  . . .E=D=D=D=D=L= . . .E=E=E=L=L=L= . . .L=L=L=L=L= . .
  M: D=D=L= . . . .E=D=D=D=D= . . .D=D=D=D=D=L= . . .D=D=L=L=
  N:  . .L=L=L=L=L=L= . . .E=D=D=D=D=L= . . . .E=D=D=L= . . .
  O:  .D=D=D=D=D=D= . . .E=E=D=D=D=D= . . . . .D=D=D=D=D= . .
  P:  . . . . .L=L= . . .L= . . . .L= . . .D=D=D=D= . . . .D=
  Q: D= . . .D=D=D= . . . . . .E= . . .E=D= . . .D= . . . .D=
  R: L= . . . .E=D= . .E= . . . . .E= . .L=L= . .L= . . . .E=
  S: L= . . . . .E= . . . .D= . . . . .D=D=D=D= . . .E= . .E=
  T:  . . . . . . . .D=D=D= . .E=E=E= . . .E=E= . . . . . .E=
Final solution (checked for feasibility): 1187
-- Cover costs  : 1102
-- Roster costs : 85
  A: D=D=D=D=D= . .D=D= . .D=D=D=D= . .E=D=D=D=D= . .E=E= . .
  B:  .E=E=E=D= . .D=D=D=D= . . .E=E=E= . .E=E=E=D=D= . .D=D=
  C:  . .E=E=L= . .D=D=L=L=L= . .D=D= . .E=E=E= . .D=D=D=L=L=
  D: E=E=E= . . . .E=E=E=E=E= . . . .E=E=E=E=E= . .E=E=E=E=E=
  E: D=D=L= . .D=D=L=L=L= . . . . .E=E=D=L=L= . .L=L=L= . . .
  F: L=L= . .E=E=E=E=E= . .E=E=E=L= . .L=L= . .E=E=E=E=E= . .
  G: E=E= . .E=E=E=E=E= . .E=E=E=E= . .E=E= . .E=E=E=E=E= . .
  H: E=E=E=E=E= . .E=D= . .D=L=L= . .D=D=D=D=L= . . .D=D= . .
  I: D=D=D=D=D= . . .E=D=D= . . .D=D= . . .E=E=E=E=D= . .E=E=
  J:  . .D=D=D=D=D= . .E=D=D=L=L= . .D=D=D= . .D=L=L=L=L= . .
  K: D=D=D=D=D= . . . . . .E=E=E=E=D= . . .E=E=E=E=E=E= . . .
  L:  . . .E=D=D=D=D=L= . . .E=E=E=D=L=L= . . .L=L=L=L=L= . .
  M: E=D=D= . . . .D=D=D=D=D= . . .D=D=D=D=D=L= . . .D=D=L=L=
  N:  . .L=L=L=L=L=L= . . .E=D=D=D=L=L= . . . .E=D=D=D=D= . .
  O: D=D=D=D=D= . . . . .E=E=D=D=D=D= . . . . . .D=D=D=D=D=D=
  P:  . . . . .L=L= . . .L= . . . .L= . . .D=D=D=D= . . . .D=
  Q: D= . . .D=D=D= . . . . . .E= . . .E=D= . . .D= . . . .D=
  R: L= . . . .E=E= . .E= . . . . .E= . .L=L= . .L= . . . .E=
  S: L= . . . .E=E= . . . .D= . . . . .D=D=D=D= . . . . . .E=
  T:  . . . . . . . .D=D=D= . .E=E=E= . . .E=E= . . . . . .E=
Final solution (checked for feasibility): 1183
-- Cover costs  : 1102
-- Roster costs : 81
